packages/gui/src/hooks/useVerifyHash.ts

```diff
@@ -1,152 +1,336 @@
-  uri: string,
-  ignoreSizeLimit: boolean,
-  metadata: any,
-  metadataError: any,
-  isPreview: boolean,
-  dataHash: string,
-): {
+export default function useVerifyHash(props: VerifyHash): {
   isValid: boolean;
   isLoading: boolean;
   error: string | undefined;
   thumbnail: any;
+  isValidationProcessed: boolean;
+  validateNFT: boolean;
+  encoding: string;
+  isBinaryHashValid: number;
 } {
+  const {
+    nft,
+    ignoreSizeLimit,
+    metadata,
+    metadataError,
+    isPreview,
+    dataHash,
+    nftId,
+    validateNFT,
+  } = props;
   const [isValid, setIsValid] = useState(false);
+  const [isValidationProcessed, setIsValidationProcessed] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState<string | undefined>();
   const [thumbnail, setThumbnail] = useState({});
+  const [isBinaryHashValid, setIsBinaryHashValid] = useState(0);
+  const [thumbCache, setThumbCache] = useLocalStorage(
+    `thumb-cache-${nftId}`,
+    {},
+  );
+  const [contentCache, setContentCache] = useLocalStorage(
+    `content-cache-${nftId}`,
+    {},
+  );
+  const [forceReloadNFT] = useLocalStorage(`force-reload-${nftId}`, false);
+
+  const uri = nft.dataUris?.[0];
 
   let lastError: any;
 
   async function validateHash(metadata: any): Promise<void> {
     let uris: string[] = [];
     let videoThumbValid: boolean = false;
-    let imageThumbValid: boolean = false;
 
     setError(undefined);
-
     setIsLoading(true);
     setIsValid(false);
 
     if (metadata.preview_video_uris && !metadata.preview_video_hash) {
       setIsLoading(false);
-      setError('missing preview_video_hash');
+      lastError = 'missing preview_video_hash';
     } else if (metadata.preview_image_uris && !metadata.preview_image_hash) {
       setIsLoading(false);
       setIsValid(false);
-      setError('missing preview_image_hash');
+      lastError = 'missing preview_image_hash';
     } else {
+      /* ================== VIDEO THUMBNAIL ================== */
       if (metadata['preview_video_uris']) {
-        uris = metadata['preview_video_uris'];
-        for (let i = 0; i < uris.length; i++) {
-          const uri = uris[i];
-          try {
-            if (!isURL(uri)) {
-              setError('Invalid URI');
-            }
-            const { data: content, encoding } = await getRemoteFileContent(
-              uri,
-              ignoreSizeLimit ? undefined : MAX_FILE_SIZE,
-            );
-            const hash = metadata['preview_video_hash'];
-            const isHashValid = isContentHashValid(content, hash, encoding);
-            if (!isHashValid) {
-              lastError = 'thumbnail hash mismatch';
+        /* if it's cached, don't try to validate hash at all */
+        if (thumbCache.video) {
+          setThumbnail({
+            video: `cached://${thumbCache.video}`,
+          });
+          setIsLoading(false);
+          videoThumbValid = true;
```

Comment:
LGTM flagged this as an unused assignment. Should it be removed, or is there additional logic that needs to be added?